### ISSUE TITLE
Keybindings interfere with i3s default keybindings

### ISSUE THREAD

**Issue by hugohabicht01** [aaronjanse/3mux#24]
Do you have any plans to add another set of keybindings, that doesn't interfere with the default i3 keybindings? Because this is pretty much unusable out of the box on i3....

**Comment by aaronjanse** [comment]
This is a tough. I'm pretty sure ctrl+arrow bindings are already used by many shells.

But I do want 3mux to work on i3. Do you have any ideas for keybindings? Maybe a vim-like escape-then-shortcut pattern?

**Comment by hugohabicht01** [comment]
I doubt that a pattern with esc would be a good idea, because many people use vim and then it would result in loads of conflicts. I would suggest further expanding the tmux compatibility and adding a config file, so that users can remap keys themselves. This would probably resolve this issue the best, because most people using i3 don't have default bindings, mostly they customized them a bit. If you have a config file you can make 3mux work together with you configuration of i3.

**Comment by aaronjanse** [comment]
I'm one or two days away from config files being implemented :+1:

**Comment by pepa65** [comment]
Configurable key bindings are a must for any kind of adoption beyond the inner circle I think. Will the config files allow for that?

**Comment by aaronjanse** [comment]
Yes. Configurability will be the focus of v0.3. I'm thinking of having a help bar like nano that can be easily disabled temporarily or disabled permanently by generating a config file (which should be *very easy* to edit).

Any thoughts?

**Comment by pepa65** [comment]
For me the help bar is something I'll immediately turn off, but that's me. I am more interested in the config file, and if it's easy to edit, great! What format were you thinking? TOML is easy, fairly small implementations in golang, but there might be much smaller ones that suffice.

**Comment by pepa65** [comment]
There is gopkg.in/go-ini/ini.v0 (and there is also a more extensive v1...)

**Comment by aaronjanse** [comment]
> TOML is easy, fairly small implementations in golang, but there might be much smaller ones that suffice.

I'd love if 3mux could use TOML. The config would probably look something like this (with lots of leniency for shortcut capitalization)

```toml
new-pane = ['Alt+N', 'Alt+Enter']

[move-pane]
up = ['Alt+Shift+Up']
down = ['Alt+Shift+Down']
left = ['Alt+Shift+Left', 'Ctrl+B {']
right = ['Alt+Shift+Right', 'Ctrl+B }']
```

If you have any suggestions, I'd love to hear.

> For me the help bar is something I'll immediately turn off, but that's me.

I definitely want to cater to people who don't want the help bar.

To generate the config, you'd either do something like `3mux --generate-config` or run `3mux` and hide the help bar (`Alt+\`?) which would prompt the user to create a config. This is my prototype so far:

![image](https://user-images.githubusercontent.com/16829510/79831713-c98e5680-835c-11ea-9441-df45658cd07e.png)

Any thoughts?

**Comment by pepa65** [comment]
Looks reasonable! I would automatically generate a config file on first run (perhaps with a commandline option for a non-default location), If you want to offer the option of not automatically create it, users can use a commandline option for that.

**Comment by aaronjanse** [comment]
> I would automatically generate a config file on first run

Oh, perfect. I'll make sure to include documentation in the (small) generated config file to make things easy.

> perhaps with a commandline option for a non-default location

Sure, and I'll make sure to use `$XDG_CONFIG_HOME` by default. Maybe hiding the prompt should simply ask the user if they want to hide it forever? I can't imagine anyone using the bar past their first few minutes of moving panes around for fun.

Thanks for the feedback!

**Comment by aaronjanse** [comment]
Is this `mode` config syntax too confusing?

```toml
add-pane  = ['Alt+N', 'Alt+Enter']
kill-pane = ['Alt+Shift+Q']

toggle-fullscreen = ['Alt+Shift+F']
toggle-search = ['Alt+/']

show-help = ['Alt+Shift+?']

move-pane-up    = ['Alt+Shift+Up',    'Alt+Shift+K']
move-pane-down  = ['Alt+Shift+Down',  'Alt+Shift+J']
move-pane-left  = ['Alt+Shift+Left',  'Alt+Shift+H']
move-pane-right = ['Alt+Shift+Right', 'Alt+Shift+L']

move-selection-up    = ['Alt+Up',    'Alt+K']
move-selection-down  = ['Alt+Down',  'Alt+J']
move-selection-left  = ['Alt+Left',  'Alt+H']
move-selection-right = ['Alt+Right', 'Alt+L']

# NAME has no meaning apart from may be displayed in a status bar
# [modes.NAME]
# mode-starter = ['KEYCODE'] # type KEYCODE to start this mode
# sticky = STICKY
# # if STICKY:
# #   stay in this mode until we see an unrecognized key
# # else:
# #   exit this mode after the first keypress
#
# # do ACTION when we see KEYCODE while in this mode
# ACTION = ['KEYCODE']

[modes.resize]
mode-starters = ['Alt+R']
sticky = true

resize-up    = ['Up',    'j']
resize-down  = ['Down',  'k']
resize-left  = ['Left',  'h']
reisze-right = ['Right', 'l']

[modes.tmux]
mode-starters = ['Ctrl+B']
sticky = false

split-pane-vert  = ['%']
split-pane-horiz = ['"']
move-pane-left   = ['{']
move-pane-right  = ['}']
```

**Comment by pepa65** [comment]
Looks great to me!

**Comment by aaronjanse** [comment]
Fixed in `v0.3.0`
